### PR TITLE
feat(web-app): enhanced mdx code component features

### LINKED ISSUE #363: Add full code snippet functionality to MDX component
Add full code component functionality to match Gatsby theme

- [ ] Path / src links
- [ ] Prism syntax highlighting

### PR DESCRIPTION
**Pull request by alisonjoseph**
Closes #363

#### Changelog

- Add `prism-react-renderer` package
- Custom "carbon" prism theme
- Custom rehype plugin to access meta information from code snippet
- Update code component to display syntax highlighting, src and path info

#### Testing / reviewing

http://localhost:3000/mdx/code

https://gatsby-theme-carbon.vercel.app/components/code-blocks

### CODE REVIEW

**Review by francinelucca — COMMENTED** [review]
looking good!
There seems to be a slight background color difference between platform and gatsby, are we cool with that?

**Comment by alisonjoseph** [comment]
Good catch, we're using the Carbon tokens, where Gastby is using a hard coded color. I'll check in with design on this to make sure.

File: services/web-app/components/code/path.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+          style={{ position: 'relative', zIndex: 200 }}
+        >
+          <Launch alt="View source" />
+        </a>
+      )}
+    </div>
+  )
+}
+
+export default Path
```
Comment (andreancardona) [inline services/web-app/components/code/path.js:33]: do we want to add proptype definitions for src and path here?

**Comment by andreancardona** [comment]
Wanted to make sure this was anticipated theme / color for the clipboard

https://user-images.githubusercontent.com/32720851/167924272-68c5f5d5-bde0-4a7d-94b1-600b0d4413a2.mov

**Comment by alisonjoseph** [comment]
Oooh good catch @andreancardona I wonder why they aren't getting the correct theme 🤔

**Comment by alisonjoseph** [comment]
@carbon-design-system/carbon-platform-designers see [questions in slack](https://ibm-studios.slack.com/archives/C03BZEUKBP1/p1652286253875419) around theme, backgrund colors and tooltip color

**Comment by ljcarot** [comment]
@alisonjoseph Check with System squad...I think this is a known bug -> https://github.com/carbon-design-system/carbon/issues/11346

**Comment by alisonjoseph** [comment]
@ljcarot thanks for the link! After looking at this again it is displaying in the correct theme I'm setting, I just need to sync with design and see if that's what we want or not.

File: services/web-app/components/code/path.js (hunk 1 above)
Comment (alisonjoseph) [reply services/web-app/components/code/path.js:41]: done!

**Comment by alisonjoseph** [comment]
Synced with design over slack, this should be ready for review now @francinelucca @andreancardona

File: services/web-app/components/code/path.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      <span className={styles.path}>{path}</span>
+
+      {src && (
+        <a
+          target="_blank"
+          rel="noopener noreferrer"
+          title="View source"
+          className={styles.icon}
+          href={src}
+          style={{ position: 'relative', zIndex: 200 }}
```
Comment (francinelucca) [inline services/web-app/components/code/path.js:25]: should we move this inline style into services/web-app/components/code/code.module.scss ?

**Review by francinelucca — DISMISSED** [review]
LGTM! minor suggestion

**Review by andreancardona — DISMISSED** [review]
🔥

File: services/web-app/components/code/path.js (hunk 2 above)
Comment (alisonjoseph) [reply services/web-app/components/code/path.js]: good catch! we don't even need those styles, removed.

**Review by francinelucca — APPROVED** [review]
💃

**Pull request merged**